### ISSUE TITLE
Add Extension point to register FluentValidation without IMvcBuilder

### ISSUE THREAD

**Issue by Sherif-Ahmed** [FluentValidation/FluentValidation#1726]
### Is your feature request related to a problem?

Not a problem, but enhancement for easy to use in some situations like OrchardCore.
In OrchardCore I don't have access to IMvcBuilder (I can work around it), but if I have an extension point so that I can call  `services.AddFluentValidation()` it will be good.

### Solution

I take a look at the code and figured out that you are just using IMvcBuilder to call `mvcBuilder.AddMvcOptions` which internally is calling `services.Configure<MvcOptions>()` so we can have a new extension point that calls directly `services.Configure<MvcOptions>()`

```
public static IServiceCollection AddFluentValidation(this IServiceCollection services, Action<FluentValidationMvcConfiguration> configurationExpression = null) {
    var config = new FluentValidationMvcConfiguration(ValidatorOptions.Global);
    configurationExpression?.Invoke(config);

    RegisterServices(services, config);

    services.Configure<MvcOptions>(options => {
        // Check if the providers have already been added.
        // We shouldn't have to do this, but there's a bug in the ASP.NET Core integration
        // testing components that can cause Configureservices to be called multple times
        // meaning we end up with duplicates.
        if (!options.ModelMetadataDetailsProviders.Any(x => x is FluentValidationBindingMetadataProvider)) {
	    options.ModelMetadataDetailsProviders.Add(new FluentValidationBindingMetadataProvider());
        }

        if (!options.ModelValidatorProviders.Any(x => x is FluentValidationModelValidatorProvider)) {
	    options.ModelValidatorProviders.Insert(0, new FluentValidationModelValidatorProvider(
	        config.ImplicitlyValidateChildProperties,
		config.ImplicitlyValidateRootCollectionElements));
        }
    });

    return services;
}
```

**Comment by JeremySkinner** [comment]
Hi, thanks for the suggestion - that seems like a reasonable addition. If you'd like to put together a pull request then I'd be happy to review it. I think the PR would need to do the following:

- [ ] Add the new overload 
- [ ] Refactor existing overloads to remove as much duplication as possible
- [ ] Add new integration test that checks that calling `services.AddFluentValidation()` prior to `services.AddMvc()` doesn't break anything

**Comment by Sherif-Ahmed** [comment]
The PR added please verify it.

**Comment by JeremySkinner** [comment]
Thanks, I'll aim to do that this evening or tomorrow.

**Comment by JeremySkinner** [comment]
Merged, and will be in the 10.2 release.
